src/ninja.cc

```diff
@@ -1087,6 +1096,124 @@ int NinjaMain::ToolRestat(const Options* options, int argc, char* argv[]) {
   return EXIT_SUCCESS;
 }
 
+struct Compdb_Targets {
+  enum class Action {
+    display_help_and_exit,
+    emit_commands
+  };
+
+  Action action;
+  EvaluateCommandMode eval_mode = ECM_NORMAL;
+
+  std::vector<StringPiece> targets;
+
+  static Compdb_Targets parse(int argc, char* argv[]) {
```

Comment:
"parse" is a bit vague here, and you would usually use such a name for something that changes the state of the current instance, not return a new one. Can you call it something like "CreateFromArgs", "NewFromArgs" or something similar.